quinn/src/endpoint.rs

```diff
@@ -408,13 +413,62 @@ impl State {
                     .write(IoSliceMut::<'a>::new(buf));
             });
         let mut iovs = unsafe { iovs.assume_init() };
+        if self.abandoned_socket.is_some() {
```

Comment:
This is a big block of code that's mostly shared with the normal path. Can we factor it out?